### ISSUE TITLE
[Bug Report] Unstable softmax 

### ISSUE THREAD

**Issue by rfurko-tt** [tenstorrent/tt-metal#12661]
**Describe the bug**
`ttnn.softmax` is unstable when numbers are large

**To Reproduce**

Example 1

```
with ttnn.manage_device(device_id=0) as device:
    x = torch.tensor([[[[1000, 1001]]]], dtype=torch.float32)
    print(x.shape)
    x_tt = ttnn.from_torch(x, layout=ttnn.TILE_LAYOUT, device=device)
    moreh_softmax_tt = ttnn.experimental.operations.primary.moreh_softmax(x_tt, dim=3)
    x_softmax_tt = ttnn.softmax(x_tt, dim=-1)
    x_correct_tt = ttnn.softmax(x_tt - ttnn.max(x_tt), dim=-1)

    print("x:", x)
    print("x_tt:", ttnn.to_torch(x_tt))
    print("moreh_softmax_tt:", ttnn.to_torch(moreh_softmax_tt))
    print("x_softmax_tt:", ttnn.to_torch(x_softmax_tt))
    print("x_correct_tt:", ttnn.to_torch(x_correct_tt))
```

Output: 
```
torch.Size([1, 1, 1, 2])
                   Test | INFO     | Small tensor algorithm selected
x: tensor([[[[1000., 1001.]]]])
x_tt: TorchTensor([[[[1000., 1001.]]]])
moreh_softmax_tt: TorchTensor([[[[0.9927, 0.0000]]]])
x_softmax_tt: TorchTensor([[[[0., 0.]]]])
x_correct_tt: TorchTensor([[[[0.2542, 0.7265]]]])
```

Example 2 (moreh softmax strange outputs):
```
with ttnn.manage_device(device_id=0) as device:
    x = torch.tensor([[[[0, 1]]]], dtype=torch.float32)
    print(x.shape)
    x_tt = ttnn.from_torch(x, layout=ttnn.TILE_LAYOUT, device=device)
    moreh_softmax_tt = ttnn.experimental.operations.primary.moreh_softmax(x_tt, dim=3)
    x_softmax_tt = ttnn.softmax(x_tt, dim=-1)
    x_correct_tt = ttnn.softmax(x_tt - ttnn.max(x_tt), dim=-1)

    print("x:", x)
    print("x_tt:", ttnn.to_torch(x_tt))
    print("moreh_softmax_tt:", ttnn.to_torch(moreh_softmax_tt))
    print("x_softmax_tt:", ttnn.to_torch(x_softmax_tt))
    print("x_correct_tt:", ttnn.to_torch(x_correct_tt))
```
output:

```
torch.Size([1, 1, 1, 2])
                   Test | INFO     | Small tensor algorithm selected
x: tensor([[[[0., 1.]]]])
x_tt: TorchTensor([[[[0., 1.]]]])
moreh_softmax_tt: TorchTensor([[[[0.4958, 0.0000]]]])
x_softmax_tt: TorchTensor([[[[0.2542, 0.7265]]]])
x_correct_tt: TorchTensor([[[[0.2790, 0.7224]]]])
```

**Expected behavior**
Provided in examples

**Please complete the following environment information:**
- OS: Ubuntu 20.04
- Version of software: a7d9204fc9cdf579e0a512e3c26fb291b398c75b

**Additional context**
Softmax is one of the key pieces of the attention mechanism.

**Comment by dmakoviichuk-tt** [comment]
hi @yugaoTT  it is a big blocker for us.
cc @jvasilje

**Comment by yugaoTT** [comment]
a stable version softmax requires to use max (reduce max) to get the max value from a row, if max has issues, not sure if it affects stable softmax. We'll figure it out

**Comment by yugaoTT** [comment]
Hey @dmakoviichuk-tt I have the fix for stable softmax here: https://github.com/tenstorrent/tt-metal/pull/13068
For the moreh versions that require this feature, please file an issue to them.

**Comment by jvasilje** [comment]
@yugaoTT @dmakoviichuk-tt has the fix been verified? why is this issue still open?
closing for now.
